Tasks/AzureRmWebAppDeployment/AzureRmWebAppDeployment.ps1

```diff
@@ -1,134 +1,111 @@
-param
-(
-    [String] [Parameter(Mandatory = $true)]
-    $ConnectedServiceName,
+[CmdletBinding()]
+param()
 
-    [String] [Parameter(Mandatory = $true)]
-    $WebAppName,
+Trace-VstsEnteringInvocation $MyInvocation
 
-    [String] [Parameter(Mandatory = $true)]
-    $DeployToSlotFlag,
+try{
 
-    [String] [Parameter(Mandatory = $false)]
-    $ResourceGroupName,
+	# Get inputs.
+	$WebAppName = Get-VstsInput -Name WebAppName -Require
+	$DeployToSlotFlag = Get-VstsInput -Name DeployToSlotFlag -Require -AsBool
+	$ResourceGroupName = Get-VstsInput -Name ResourceGroupName
+	$SlotName = Get-VstsInput -Name SlotName
+	$Package = Get-VstsInput -Name Package -Require
+	$RemoveAdditionalFilesFlag = Get-VstsInput -Name RemoveAdditionalFilesFlag -AsBool
+	$ExcludeFilesFromAppDataFlag = Get-VstsInput -Name ExcludeFilesFromAppDataFlag -AsBool
+	$TakeAppOfflineFlag = Get-VstsInput -Name TakeAppOfflineFlag -AsBool
+	$VirtualApplication = Get-VstsInput -Name VirtualApplication 
+	$AdditionalArguments = Get-VstsInput -Name AdditionalArguments 
+	$WebAppUri = Get-VstsInput -Name WebAppUri
+	$SetParametersFile = Get-VstsInput -Name SetParametersFile
 
-    [String] [Parameter(Mandatory = $false)]
-    $SlotName,
+	# Initialize Azure.
 
-    [String] [Parameter(Mandatory = $true)]
-    $Package,
-    
-    [String] [Parameter(Mandatory = $false)]
-    $SetParametersFile,
+	Import-Module $PSScriptRoot\ps_modules\VstsAzureHelpers_
+	Initialize-Azure
 
-    [String] [Parameter(Mandatory = $false)]
-    $RemoveAdditionalFilesFlag,
+	# Import the loc strings.
+	Import-VstsLocStrings -LiteralPath $PSScriptRoot/Task.json
 
-    [String] [Parameter(Mandatory = $false)]
-    $ExcludeFilesFromAppDataFlag,
+	$WebAppUri = $WebAppUri.Trim()
+	$Package = "$Package".Trim('"').Trim()
 
-    [String] [Parameter(Mandatory = $false)]
-    $TakeAppOfflineFlag,
-
-    [String] [Parameter(Mandatory = $false)]
-    $VirtualApplication,
-
-	[String] [Parameter(Mandatory = $false)]
-	[String] $AdditionalArguments,
-
-    [String] [Parameter(Mandatory = $false)]
-    [string]$WebAppUri
-)
+	if( [string]::IsNullOrEmpty($Package) ){
+		Throw (Get-VstsLocString -Key "Invalidwebapppackagepathprovided")
+	}
 
-Write-Verbose "Starting AzureRM WebApp Deployment Task"
+	$SetParametersFile = $SetParametersFile.Trim('"').Trim()
```

Comment:
I had a comment here before about the possibility of null ref exception. Re-adding in case you overlooked it by accident. Again, a safer option would be the following (double quotes added around the variable before calling trim): 
```powershell
$SetParametersFile = "$SetParametersFile".Trim('"').Trim()
```